tests/unit/via/services/youtube_transcript_test.py

```diff
@@ -0,0 +1,300 @@
+from via.exceptions import UnhandledUpstreamException
+from via.services.youtube_transcript import (
+    TranscriptInfo,
+    YouTubeTranscriptService,
+    factory,
+)
+
+
+class TestTranscriptInfo:
+    @pytest.mark.parametrize(
+        "transcript_info,expected_id",
+        [
+            (
+                TranscriptInfoFactory(),
+                "en-us..RW5nbGlzaCAoVW5pdGVkIFN0YXRlcyk=",
+            ),
+            (
+                TranscriptInfoFactory(autogenerated=True),
+                "en-us.a.RW5nbGlzaCAoVW5pdGVkIFN0YXRlcyk=",
+            ),
+        ],
+    )
+    def test_id(self, transcript_info, expected_id):
+        assert transcript_info.id == expected_id
+
+
+class TestYouTubeTranscriptService:
+    def test_get_transcript_infos(self, svc, http_service):
+        # The JSON response body from the YouTube API.
+        response_json = {
+            "captions": {
+                "playerCaptionsTracklistRenderer": {
+                    "captionTracks": [
+                        {
+                            "languageCode": "en",
+                            "name": {"simpleText": "English"},
+                            "baseUrl": "https://example.com/transcript_1",
+                        },
+                        {
+                            "languageCode": "en-us",
+                            "name": {"simpleText": "English (United States)"},
+                            "baseUrl": "https://example.com/transcript_2",
+                        },
+                    ]
+                }
+            }
+        }
+        response = http_service.post.return_value = Response()
+        response.raw = BytesIO(json.dumps(response_json).encode("utf-8"))
+
+        transcript_infos = svc.get_transcript_infos("test_video_id")
+
+        http_service.post.assert_called_once_with(
+            "https://youtubei.googleapis.com/youtubei/v1/player",
+            json={
+                "context": {
+                    "client": {
+                        "hl": "en",
+                        "clientName": "WEB",
+                        "clientVersion": "2.20210721.00.00",
+                    }
+                },
+                "videoId": "test_video_id",
+            },
+        )
+        caption_tracks = response_json["captions"]["playerCaptionsTracklistRenderer"][
+            "captionTracks"
+        ]
+        assert transcript_infos == [
+            Any.instance_of(TranscriptInfo).with_attrs(
+                {
+                    "language_code": caption_tracks[0]["languageCode"],
+                    "autogenerated": False,
+                    "name": caption_tracks[0]["name"]["simpleText"],
+                    "url": caption_tracks[0]["baseUrl"],
+                }
+            ),
+            Any.instance_of(TranscriptInfo).with_attrs(
+                {
+                    "language_code": caption_tracks[1]["languageCode"],
+                    "autogenerated": False,
```

Comment:
I should make one of these test transcripts be an auto-generated one so that we're covering that code path